File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.java
class Solution {
    private long[] dp = new long[100001];
    
    public int solution(int n) {
        dp[0] = 0;
        dp[1] = 1;
        dp[2] = 1;
        for (int i = 3; i < 100001; i++) {
            dp[i] = dp[i - 1] + dp[i - 2];
            dp[i] %= 1234567;
        }
        return (int) dp[n];
    }
}

File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.py
dp = [0] * (100001)

def solution(n):
    dp[0] = 0
    dp[1] = 1
    dp[2] = 1
    for i in range(3, 100001):
        dp[i] = dp[i - 1] + dp[i - 2]
    answer = dp[n] % 1234567
    return answer